src/components/sections/ProductGallery/ProductGallery.tsx

```diff
@@ -38,36 +41,40 @@ function ProductGallery({ title, slug }: Props) {
   return (
     <>
       <div className="product-listing__filters">
-        <Filter
-          slug={slug}
-          isOpen={isFilterOpen}
-          facets={orderedFacets}
-          onDismiss={() => setIsFilterOpen(false)}
-        />
+        <SkeletonFilter loading={orderedFacets?.length === 0}>
```

Comment:
Agree with @danzanzini. The first approach looks better to me.

Reply:
for this case, I think that is better to keep the skeleton outside because the Filter itself already has too much logic involving the `orderedFacets` results